modules/utxo-lib/test/integration_local_rpc/parse.ts

```diff
@@ -74,34 +90,83 @@ function runTestParse(network: Network, txType: FixtureTxType, scriptType: Scrip
       return;
     }
 
-    it(`verifySignatures`, function () {
+    function getPrevOutputValue(input: { txid?: string; hash?: Buffer; index: number }) {
+      if (input.hash) {
+        input = {
+          ...input,
+          txid: getTxidFromHash(input.hash),
+        };
+      }
+
+      const inputTx = fixture.inputs.find((tx) => tx.txid === input.txid);
+      if (!inputTx) {
+        throw new Error(`could not find inputTx`);
+      }
+      const prevOutput = inputTx.vout[input.index];
+      if (!prevOutput) {
+        throw new Error(`could not prevOutput`);
+      }
+      return prevOutput.value * 1e8;
+    }
+
+    it(`verifySignatures for original transaction`, function () {
       parsedTx.ins.forEach((input, i) => {
-        const inputTxid = Buffer.from(input.hash).reverse().toString('hex');
-        const inputTx = fixture.inputs.find((tx) => tx.txid === inputTxid);
-        if (!inputTx) {
-          throw new Error(`could not find inputTx`);
-        }
-        const prevOutput = inputTx.vout[input.index];
-        if (!prevOutput) {
-          throw new Error(`could not find prevOutput`);
-        }
+        const prevOutValue = getPrevOutputValue(input);
         const { publicKeys } = parseSignatureScript(input);
         if (!publicKeys) {
           throw new Error(`expected publicKeys`);
         }
         assert.strictEqual(publicKeys.length, 3);
         publicKeys.slice(0, 2).forEach((publicKey) => {
           assert.strictEqual(
-            verifySignature(parsedTx, i, prevOutput.value * 1e8, {
+            verifySignature(parsedTx, i, prevOutValue, {
               publicKey,
             }),
             true
           );
 
-          assert.strictEqual(verifySignature(parsedTx, i, prevOutput.value * 1e8), true);
+          assert.strictEqual(verifySignature(parsedTx, i, prevOutValue), true);
+        });
+      });
+    });
+
+    function getRebuiltTransaction(signKeys?: bip32.BIP32Interface[]) {
+      assert.strict(parsedTx.outs.length === 1);
+      const recipientScript = parsedTx.outs[0].script;
+      return createSpendTransactionFromPrevOutputs(
+        fixtureKeys,
+        scriptType,
+        parsedTx.ins.map((i) => [getTxidFromHash(i.hash), i.index, getPrevOutputValue(i)]),
+        recipientScript,
+        network,
+        { signKeys }
+      ) as unknown as Transaction;
```

Comment:
Doesn't `createSpendTransactionFromPrevOutputs` return a `Transaction`?

Reply:
It was a slightly different type (for no good reason). Consolidated.